### PR TITLE
ENH fix the sysroot shims

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/install_ld.sh [hunk 1]
```diff
@@ -8,10 +8,9 @@ make DESTDIR=$PWD/install install-strip
...
 OLD_CHOST="${ctng_cpu_arch}-${ctng_vendor}-linux-gnu"
+
 mkdir -p $PREFIX/bin
-mkdir -p $PREFIX/$OLD_CHOST/bin
-mkdir -p $PREFIX/$CHOST/bin
 cp $PWD/install/$PREFIX/bin/$CHOST-ld $PREFIX/bin/$CHOST-ld
+
 ln -s $PREFIX/bin/$CHOST-ld $PREFIX/bin/$OLD_CHOST-ld
-ln -s $PREFIX/bin/$CHOST-ld $PREFIX/$OLD_CHOST/bin/ld
 ln -s $PREFIX/bin/$CHOST-ld $PREFIX/$CHOST/bin/ld
```
Comment (isuruf) [inline recipe/install_ld.sh:16]:
```suggestion
```

File: recipe/install_binutils.sh [hunk 2]
```diff
@@ -5,20 +5,20 @@ set -e
...
 done
 
-rm $PREFIX/bin/$HOST-ld || true;
-rm $PREFIX/bin/$OLD_HOST-ld || true;
-rm $PREFIX/$OLD_HOST/bin/ld || true;
-rm $PREFIX/$HOST/bin/ld || true;
+rm $PREFIX/bin/$CHOST-ld || true;
+rm $PREFIX/bin/$OLD_CHOST-ld || true;
+rm $PREFIX/$OLD_CHOST/bin/ld || true;
+rm $PREFIX/$CHOST/bin/ld || true;
```
Comment (isuruf) [inline recipe/install_binutils.sh:22]:
```suggestion
```

File: recipe/install_binutils.sh [hunk 3]
```diff
@@ -5,20 +5,20 @@ set -e
...
 make install-strip
-export HOST="${ctng_cpu_arch}-conda-linux-gnu"
-export OLD_HOST="${ctng_cpu_arch}-${ctng_vendor}-linux-gnu"
-mkdir -p $PREFIX/$OLD_HOST/bin
+export CHOST="${ctng_cpu_arch}-conda-linux-gnu"
+export OLD_CHOST="${ctng_cpu_arch}-${ctng_vendor}-linux-gnu"
+mkdir -p $PREFIX/$OLD_CHOST/bin
+
 # Remove hardlinks and replace them by softlinks
 for tool in addr2line ar as c++filt dwp elfedit gprof ld.bfd ld.gold nm objcopy objdump ranlib readelf size strings strip; do
```
Comment (isuruf) [inline recipe/install_binutils.sh:13]:
```suggestion
for tool in addr2line ar as c++filt dwp elfedit gprof ld ld.bfd ld.gold nm objcopy objdump ranlib readelf size strings strip; do
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -37,7 +37,9 @@ outputs:
     requirements:
       build:
       host:
+        - sysroot_{{ target_platform }} {{ sysroot_ver }} conda_variant_*
       run:
+        - sysroot_{{ target_platform }}
```
Comment (isuruf) [inline recipe/meta.yaml:42]:
```suggestion
```

File: recipe/install_ld.sh (hunk 1 above)
Comment (isuruf) [inline recipe/install_ld.sh:16]:
```suggestion
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -37,7 +37,9 @@ outputs:
     requirements:
       build:
       host:
+        - sysroot_{{ target_platform }} {{ sysroot_ver }} conda_variant_*
```
Comment (isuruf) [inline recipe/meta.yaml:40]:
```suggestion
```

**Comment by beckermr** [comment]
Conda is not installing the test env properly because the test use the version of gcc without the sysroot.

**Comment by beckermr** [comment]
I think we need to either mark the 7.50 packages as broken right now. This version of binutils is not meant to be backwards compatible anyways.

**Comment by beckermr** [comment]
I test these builds locally with version `*_8` of the compilers (the latest build we marked as broken). They work fine. So I think this PR is good to merge once we mark the current 7.5.0 build as broken.

**Comment by beckermr** [comment]
This one is no longer needed.

**Pull request closed without merging**